py/desispec/workflow/queue.py

```diff
@@ -452,3 +453,132 @@ def any_jobs_failed(statuses, failed_states=None):
     if failed_states is None:
         failed_states = get_failed_states()
     return np.any([status in failed_states for status in statuses])
+
+def get_jobs_in_queue(user=None, include_scron=False, dry_run_level=0):
+    """
+    Queries the NERSC Slurm database using sacct with appropriate flags to get
+    information about specific jobs based on their jobids.
+
+    Parameters
+    ----------
+    user : str
+        NERSC user to query the jobs for
+    include_scron : bool
+        True if you want to include scron entries in the returned table.
+        Default is False.
+    dry_run_level : int
+        Whether this is a simulated run or real run. If nonzero, it is a
+        simulation and it returns a default table that doesn't query the
+        Slurm scheduler.
+
+    Returns
+    -------
+    Table
+        Table with the columns JOBID, PARTITION, NAME, USER, ST, TIME, NODES,
+        NODELIST(REASON) for the specified user.
+    """
+    log = get_logger()
+
+    cmd = f'squeue -u {user} -o "%i,%P,%j,%u,%t,%M,%D,%R"'
```

Comment:
let's intercept the default `user==None` case and set `user=os.environ['USER']`.

Reply:
Done